bgcval2/bgcvaltools/removeMaskFromshelve.py

```diff
@@ -31,27 +31,25 @@
 
 """
 
-from shelve import open as shopen
+from shelve import open as shOpen
 from glob import glob
 
 
 def removeFromShelves(fn, removeRegions):
     print('removing:', removeRegions, 'from', fn)
-    sh = shopen(fn)
-
-    modeldata = sh['modeldata']
-
-    for key in list(modeldata.keys()):
-        try:
-            (r, l, m) = key
-        except:
-            continue
-        if r in removeRegions:
-            print('modeldata[', (r, l, m), '] will be deleted')
-            del modeldata[(r, l, m)]
-
-    sh['modeldata'] = modeldata
-    sh.close()
+    with shOpen(fn) as sh:
+        modeldata = sh['modeldata']
+        for key in list(modeldata.keys()):
+            try:
+                (r, l, m) = key
+            except ValueError:
+                continue
+            if r in removeRegions:
+                print('modeldata[', (r, l, m), '] will be deleted')
+                del modeldata[(r, l, m)]
+
+        sh['modeldata'] = modeldata
```

Comment:
could the del above remove *all* `modeldata` instances hence this will return a `NameError`?